ui/components/Lifecycle/Environments/index.js

```diff
@@ -385,15 +385,7 @@ const Environments = ({ organization, classes }) => {
   const handleAssignConnectionData = (updatedAssignedData) => {
     const { addedConnectionsIds, removedConnectionsIds } =
       getAddedAndRemovedConnection(updatedAssignedData);
-    (addedConnectionsIds.length > 0 || removedConnectionsIds.length) > 0 &&
-    (CAN(
-      keys.ASSIGN_CONNECTIONS_TO_ENVIRONMENT.action,
-      keys.ASSIGN_CONNECTIONS_TO_ENVIRONMENT.subject,
-    ) ||
-      CAN(
-        keys.REMOVE_CONNECTIONS_FROM_ENVIRONMENT.action,
-        keys.REMOVE_CONNECTIONS_FROM_ENVIRONMENT.subject,
-      ))
```

Comment:
@sudhanshutech why are you removing keys

Reply:
because the transferlist we use already holds the ability to add permissions(which checks to assign or not) and we are using it so we don't required here duplicative thing